### ISSUE TITLE
Suggestion: Commercial support licencing options

### ISSUE THREAD

**Issue by PhilipOakley** [CoatiSoftware/Sourcetrail#810]
At one of my former companies, they were loathe to use open source because it was 'unsupported'. In some areas RHEL was allowed and paid for, because it gave a nice commercial 'hand-off' of responsibility for support. Only 'bought/supported' software was allowed as mainstream code.

Can I suggest that [Croati] Sourcetrail also be offered with a paid for support licence, as a complement to Patreon, so that users in larger corporate organisations can make it feel like they are 'buying' the software [support]. 

One could offer initial 30 day or 15 day trial support licences (just like big fee software gets 'offered'), which can then be purchased on monthly / quarterly  /annual terms for number of 'seats' of support.

This will fit well with some corporates and allow [open source] users to be pro-active in their company environments, and at the same time create the cash flow mentioned in the [Sourcetrail blog announcement](https://www.sourcetrail.com/blog/open_source/).

This would be an exercise in branding, and be fully in line with the GPL, and lets users ensure that their corporate environment support the open source development, and the open source suppliers support their predicament.

**Comment by egraether** [comment]
During our open-source announcement we have seen suggestions like this in multiple comment sections. From our experience with commercial licensing in the past years, I have some doubts whether this would work for us:
* Sourcetrail is a productivity tool. If there is a problem with it, developers can still continue to work. Managers are less inclined to spend money on something that is, at least from their point of view, not necessary.
* Customers that pay for support expect it within 24h. If we have another full-time employment, we simply don't have the time and capacity to respond to support requests within a timely manner.
* Support requests often contain project setup issues. While there are cases that can be resolved by explaining how the project needs to be configured, there are also cases where Sourcetrail needs to be extended to make it work. This results in many hours of work and customers expect to receive the changed version right away and don't want to wait for a new release, so we constantly have to release maintenance builds as well.

I'm not saying that it cannot work, but it would be a lot of effort for us. To make it financially viable these support licenses need to be pretty expensive, which in turn would further decrease the target audience. I would expect that we end up with just a few customers.

But that's just my two cents on the issue. We will track the demand for this.

**Comment by PhilipOakley** [comment]
I'd agree about many of the potential points regarding 'demanding' customers. 

In this case I was deliberately thinking of a slightly dumbed down 'licence' that essentially only gives the same level of support that you already give to the open source community, except perhaps they get a separate reporting address so their troubles can be anonymised for the issues list,  but mainly it allows users, whose managements are being a pain, to pay for a licence so that procedural boxes are ticked.

Mainly it's about those that want to pay, because it's their only way to use the code.

Basically charge a Patreon supporter fee for the privilege of Croati still being their to support the code, but framed in procedural business support terms.

**Comment by PhilipOakley** [comment]
>  I was deliberately thinking of a slightly dumbed down 'licence' that essentially

I forgot the term "Shrink-wrapped", so that it wasn't something that was negotiated, rather it was something (any-minimal-thing) that you'd be able to cope with. If some customers need to pay, let them pay ;-)
